### ISSUE TITLE
incorrect-bytes-format accepts a truncated sequence with a trailing dash (AB-)

### ISSUE THREAD

**Issue by VasilevNStas** [objectionary/lints#1383]
## Problem

`incorrect-bytes-format` accepts a truncated byte sequence with a trailing dash (e.g. `AB-`) as valid.

## Root cause

`src/main/resources/org/eolang/lints/critical/incorrect-bytes-format.xsl:18`:

```xml
<xsl:variable name="bytes" select="'^(--|[0-9A-F]{2}(-|(-[0-9A-F]{2})+))$'"/>
```

The regex allows a sequence `[0-9A-F]{2}` followed by a bare `-` alternative: `AB-` matches (`AB` + `-`). A trailing dash is a malformed byte sequence semantically.

## Minimal example

```eo
+package x
+architect yegor@256.com
+version 0.0.1
+home https://github.com/objectionary
+spdx SPDX-FileCopyrightText: Copyright (c) 2026 Objectionary.com
+spdx SPDX-License-Identifier: MIT

[] > data
  AB- > @
```

No `incorrect-bytes-format` defect is reported for `AB-`.

## Expected behavior

A dash must always be *between* two byte groups; trailing (and leading) dashes should be rejected. Example fix: `^(--|[0-9A-F]{2}(-[0-9A-F]{2})*)$`.

## Related

- `critical/bytes-without-data.xsl` — sibling data-bytes rule.
- Existing packs under `packs/single/incorrect-bytes-format/`.

**Comment by gemshrine** [comment]
Withdrawing this, it is not a defect. A trailing dash on a single group is how EO writes a one-byte value: `bool.eo:22` holds `FF-` and `i8.eo:19` holds `00-`, and the runtime is full of them. The `(-|(-[0-9A-F]{2})+)` alternative is there to accept exactly that, so `AB-` has to pass.

The regex is consistent about the rest of the shape. Run against the sheet:

| bytes | result |
| --- | --- |
| `AB-` | passes, one byte |
| `--` | passes, no bytes |
| `FF-11-` | defect |
| `AB` | defect |

The change I suggested would have made every one-byte literal in `eo-runtime` a critical defect. Sorry for the noise.

**Comment by gemshrine** [comment]
Withdrawing this, it is not a defect. A trailing dash on a single group is how EO writes a one-byte value: `bool.eo:22` holds `FF-` and `i8.eo:19` holds `00-`, and the runtime is full of them. The `(-|(-[0-9A-F]{2})+)` alternative is there to accept exactly that, so `AB-` has to pass.

The regex is consistent about the rest of the shape. Run against the sheet:

| bytes | result |
| --- | --- |
| `AB-` | passes, one byte |
| `--` | passes, no bytes |
| `FF-11-` | defect |
| `AB` | defect |

The change I suggested would have made every one-byte literal in `eo-runtime` a critical defect. Sorry for the noise.

**Comment by VasilevNStas** [comment]
Thanks @gemshrine for the correction — you are right. The trailing dash is the EO convention for a one-byte literal (`FF-`, `00-` in the runtime), and the existing pack `incorrect-bytes-format/catches.yaml` already treats the bare `AB` (no dash) as a defect while `AB-` is valid. My report was wrong; closing.

**Comment by 0crat** [comment]
@vasilevnstas Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+818](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
